parsil/src/assembler.rs

```diff
@@ -413,30 +415,15 @@ impl<'a, C: ContextProvider> Assembler<'a, C> {
                         } else {
                             right
                         };
-                        let bound = Some(self.expression_to_boundary(right)?);
-
-                        if bounds.high.is_some() {
-                            match bound.partial_cmp(&bounds.high) {
-                                Some(Ordering::Less) => bounds.high = bound,
-                                // impossible to say which is lower between two
-                                // conflicting high bounds
-                                Some(_) => {}
-                                None => bounds.low = None,
-                            }
-                        } else {
-                            bounds.high = bound;
-                        }
+                        let bound = self.expression_to_boundary(right)?;
+
+                        maybe_replace_min(&mut bounds.low, bound);
                     }
                     // $sid = x
                     BinaryOperator::Eq => {
-                        let bound = Some(self.expression_to_boundary(right)?);
-                        if bounds.low.is_some() && bounds.high.is_some() {
-                            bounds.low = None;
-                            bounds.high = None;
-                        } else {
-                            bounds.low = bound.clone();
-                            bounds.high = bound;
-                        }
+                        let bound = self.expression_to_boundary(right)?;
+                        maybe_replace_min(&mut bounds.high, bound.clone());
+                        maybe_replace_max(&mut bounds.low, bound);
```

Comment:
In case of equality I think we can always set `bounds.high = bound` and `bounds.low = bound`, given that equality should always be more restrictive than any other bound?

Reply:
That's right